src/main/java/cubicchunks/server/chunkio/async/forge/AsyncWorldIOExecutor.java

```diff
@@ -275,8 +280,10 @@ public static void tick() {
 		while (columnIter.hasNext()) {
 			AsyncColumnIOProvider task = columnIter.next();
 			if (task.isFinished()) {
-				if (task.hasCallbacks())
-					task.runSynchronousPart();
+				if (task.hasCallbacks()) {
+					task.runCallbacks();
+				}
+				task.runSynchronousPart();
```

Comment:
Is there any particular reason why you split of the callbacks from runSynchronousPart()? You still seem to call them together in every location